services/provider/server/server.go

```diff
@@ -596,5 +526,119 @@ func (s *OCSProviderServer) RevokeStorageClassClaim(ctx context.Context, req *pb
 
 // GetStorageClassClaim RPC call to get the ceph resources for the StorageclassClaim.
 func (s *OCSProviderServer) GetStorageClassClaimConfig(ctx context.Context, req *pb.StorageClassClaimConfigRequest) (*pb.StorageClassClaimConfigResponse, error) {
-	return &pb.StorageClassClaimConfigResponse{}, nil
+	consumerClaimObj, err := s.storageClassClaimManager.Get(ctx, req.StorageConsumerUUID, req.StorageClassClaimName)
+	if err != nil {
+		if kerrors.IsNotFound(err) {
+			return nil, status.Errorf(codes.NotFound, "failed to get storage class claim config %q for %q. %v", req.StorageClassClaimName, req.StorageConsumerUUID, err)
+		}
+		return nil, status.Errorf(codes.Internal, "failed to get storage class claim config %q for %q. %v", req.StorageClassClaimName, req.StorageConsumerUUID, err)
+	}
```

Comment:
This will not work as the errors return from ` s.storageClassClaimManager.Get` are custom ones.
I actually think that analyzing the get error fits more here and as such I will suggest just returning the underlying get error from the manager, as the manager is just an abstraction above k8s client.

Reply:
As we are returning an error with `%w` from the `s.storageClassClaimManager.Get` IMO it should work as `kerrors.IsNotFound` calls `errors.As` to check the wrapped error message. returning plain k8s error from the `Get` call so that we can return custom errors from here as you said manager is just an abstration.